packages/fuels/tests/scripts/script_assert_eq/src/main.sw

```diff
@@ -0,0 +1,59 @@
+script;
+
+// TODO: Remove when bug is fixed.
+use core::ops::*;
+fn assert_eq<T>(v1: T, v2: T)
+where
+    T: Eq
+{
+    let FAILED_ASSERT_EQ_SIGNAL = 0xffff_ffff_ffff_0003;
+    if (v1 != v2) {
+        log(v1);
+        log(v2);
+        revert(FAILED_ASSERT_EQ_SIGNAL);
+    }
+}
+// link: https://github.com/FuelLabs/sway/issues/3946
```

Comment:
Can you open a `fuels-rs` issue and link this sway issue, so we know when to tackle this TODO?

Reply:
I wanted to wait for the compiler team to fix the bug, but it seems that it could take some time. I propose we merge this with a new issue to remove the added code when the bug is fixed.